src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/api/controllers/AssessmentNeedsController.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.assessrisksandneeds.api.controllers
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.AssessmentNeedsDto
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.services.AssessmentNeedsService
+
+@RestController
+class AssessmentNeedsController(private val assessmentNeedsService: AssessmentNeedsService) {
+
+  @RequestMapping(path = ["/needs/crn/{crn}"], method = [RequestMethod.GET])
+  @Operation(description = "Gets criminogenic needs for crn")
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "403", description = "Unauthorized"),
+      ApiResponse(responseCode = "404", description = "CRN Not Found"),
+      ApiResponse(responseCode = "200", description = "OK")
+    ]
+  )
+  @PreAuthorize("hasAnyRole('ROLE_PROBATION', 'ROLE_CRS_PROVIDER')")
```

Comment:
is it a requirement to get needs for `ROLE_CRS_PROVIDER`, they normally get a more restrictive view of the data for risk

Reply:
I've removed it, I didn't realise how the two roles differed!